rpc/internal/collector.service.go

```diff
@@ -78,6 +79,10 @@ func (s *collectorService) StreamEvents(srv PoplarEventCollector_StreamEventsSer
 	for {
 		event, err := srv.Recv()
 		if err == io.EOF {
+			if err = group.closeStream(streamID); err != nil {
```

Comment:
Is there a reason we only close/flush once the stream is successfully ended? It seems like this would fail to flush all the remaining events if at least one stream returns in an error case, since the errored streams will never get closed and deleted from the streams in `flush()`.

Reply:
I think that behavior is fine, I think if something fails to persist users don't really want that data (usually if something fails to write to a collector it's because of a schema issue and the data itself is invalid).